internal/controllers/inventory_controller.go

```diff
@@ -721,6 +723,44 @@ func (t *reconcilerTask) storeSearchURL(ctx context.Context) error {
 	return nil
 }
 
+// checkForPodReadyStatus checks for all server pods to be ready.  If any pod is not yet ready, the reconciler timer is
+// set to a short value so that we can try again quickly; otherwise either an error is returned without a reconciler
+// timer to indicate that an exponential backoff is required
+func (t *reconcilerTask) checkForPodReadyStatus(ctx context.Context) (nextReconcile ctrl.Result, err error) {
```

Comment:
Consider updating ` (nextReconcile ctrl.Result, err error) {` to `(ctrl.Result, error) {`....like we are returning explicitly so might as well (i think you simply need `err := t.client.` in L738.) 

Go's "naked" return func is probably my biggest gripe with language (and widely considered a practice)! Like I find almost impossible to read when there's "return" but nothing specified especially when they go over 3 lines lol and unfortunately we have a quite bit of it in the controllers 😖

Reply:
ya, i agree... it makes it really tough to read!

Reply:
fixed